components/BottomNavigation/src/private/MDCBottomNavigationItemView.m

```diff
@@ -214,26 +213,30 @@ - (void)layoutSubviews {
 }
 
 - (void)centerLayoutAnimated:(BOOL)animated {
+  CGRect contentBoundingRect = UIEdgeInsetsInsetRect(self.bounds, self.contentInsets);
+  CGFloat centerY = CGRectGetMidY(contentBoundingRect);
+  CGFloat centerX = CGRectGetMidX(contentBoundingRect);
+  UIUserInterfaceLayoutDirection layoutDirection = self.mdf_effectiveUserInterfaceLayoutDirection;
+  if (layoutDirection == UIUserInterfaceLayoutDirectionRightToLeft) {
+    centerX = CGRectGetWidth(self.bounds) - centerX;
+  }
+
   if (self.titleBelowIcon) {
-    CGPoint iconImageViewCenter =
-        CGPointMake(CGRectGetMidX(self.bounds), CGRectGetHeight(self.iconImageView.bounds) / 2 +
-                    kMDCBottomNavigationItemViewItemInset);
-    BOOL titleVisibilityNever = self.selected &&
-        self.titleVisibility == MDCBottomNavigationBarTitleVisibilityNever;
-    BOOL titleVisibilitySelectedNever = !self.selected &&
-        (self.titleVisibility == MDCBottomNavigationBarTitleVisibilitySelected ||
-         self.titleVisibility == MDCBottomNavigationBarTitleVisibilityNever);
-    if (titleVisibilityNever) {
-      iconImageViewCenter = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
-    } else if (titleVisibilitySelectedNever) {
-      iconImageViewCenter = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
+    BOOL titleHidden =
+        self.titleVisibility == MDCBottomNavigationBarTitleVisibilityNever ||
+        (self.titleVisibility == MDCBottomNavigationBarTitleVisibilitySelected && !self.selected);
+    CGFloat iconHeight = CGRectGetHeight(self.iconImageView.bounds);
+    CGFloat labelHeight = CGRectGetHeight(self.label.bounds);
+    CGFloat totalContentHeight = iconHeight;
+    if (!titleHidden) {
+      totalContentHeight += labelHeight + self.contentVerticalMargin;
     }
+    CGPoint iconImageViewCenter =
+        CGPointMake(centerX, centerY - totalContentHeight / 2 + iconHeight / 2);
     CGPoint badgeCenter =
         CGPointMake(CGRectGetMidX(self.bounds) + CGRectGetWidth(self.iconImageView.bounds) / 2,
                     iconImageViewCenter.y - CGRectGetMidX(self.iconImageView.bounds));
```

Comment:
I know it's not a change in this PR, but shouldn't this be `CGRectGetMidY`?

Reply:
Nice catch, I think you're right